liquid-compiler/src/grammar.pest

```diff
@@ -0,0 +1,56 @@
+WHITESPACE = _{" "}
+LINE_OR_SPACE = _{ NEWLINE | WHITESPACE }
+LiquidFile = ${ SOI ~ Element* ~ EOI }
+
+// Element-level parsing
+Element = _{ Expression | Tag | Raw } 
+
+TagStart = _{ (LINE_OR_SPACE* ~ "{%-") | "{%" }
+TagEnd =   _{ ("-%}" ~ LINE_OR_SPACE*) | "%}" }
+ExpressionStart = _{ (LINE_OR_SPACE* ~ "{{-") | "{{" }
+ExpressionEnd =   _{ ("-}}" ~ LINE_OR_SPACE*) | "}}" }
+
+Tag = !{ TagStart ~ Identifier ~ TagToken* ~ TagEnd }
+Expression = !{ ExpressionStart ~ FilterChain ~ ExpressionEnd }
+// Not allowing Tag/Expression Start/End might become a problem 
+// for {% raw %} tag: unclosed delimiters won't be accepted
+Raw = @{ (!(TagStart | ExpressionStart | TagEnd | ExpressionEnd) ~ ANY)+ }              
+
+
+// Inner parsing        
+Identifier = @{ (ASCII_ALPHA | "_")+ }
+
+Variable = { Identifier 
+           ~ ( ("." ~ Identifier)
+             | ("[" ~ Value ~ "]")
+             )* 
+           }
+Value = { Literal | Variable }
+Filter = { Identifier ~ (":" ~ Value ~ ("," ~ Value)*)? } 
+FilterChain = { Value ~ ("|" ~ Filter)* } // Make * into + , filterchains without filters will degenerate to Value
+
+
+// Literals 
+StringLiteral = @{ ("'" ~ (!"'" ~ ANY)* ~ "'")
+				 | ("\"" ~ (!"\"" ~ ANY)* ~ "\"") }
+                 
+IntegerLiteral = @{ ASCII_DIGIT+ }
+FloatLiteral = @{ ASCII_DIGIT+ ~ "." ~ ASCII_DIGIT+ }
+
+BooleanLiteral = @{ "true" | "false" }
+
+Literal = { StringLiteral | IntegerLiteral | FloatLiteral | BooleanLiteral }
```

Comment:
Looks like `nil` and `null` are supported as literals
https://github.com/Shopify/liquid/blob/master/lib/liquid/expression.rb

Reply:
I'm gonna mark this as resolved as there is an issue opened for it.